src/features/wallet/scan/ScanView.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState } from 'react'
+import { useNavigation } from '@react-navigation/native'
+import { useAsync } from 'react-async-hook'
+import { StyleSheet } from 'react-native'
+import { BarCodeScanner, BarCodeScannerResult } from 'expo-barcode-scanner'
+import BottomSheet from 'react-native-holy-sheet'
+import { useSafeAreaInsets } from 'react-native-safe-area-context'
+import { Address } from '@helium/crypto-react-native'
+import Box from '../../../components/Box'
+import Text from '../../../components/Text'
+import Crosshair from './Crosshair'
+import { wp } from '../../../utils/layout'
+import Close from '../../../assets/images/close.svg'
+import TouchableOpacityBox from '../../../components/TouchableOpacityBox'
+import { triggerNavHaptic, triggerNotification } from '../../../utils/haptic'
+import sleep from '../../../utils/sleep'
+
+type Props = {
+  fromSend: boolean
+}
+
+const ScanView = ({ fromSend = false }: Props) => {
+  const [scanned, setScanned] = useState<boolean>(false)
+  const navigation = useNavigation()
+
+  const { result: permissions } = useAsync(
```

Comment:
This is cool, first time seeing the `useAsync` should we handle the error if they deny the permission?

Reply:
Ah I see below, I guess its null and not an error?